front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -45,9 +50,51 @@ const ManchetteWithSpaceTimeChartWrapper = ({
 
   const [waypointsPanelIsOpen, setWaypointsPanelIsOpen] = useState(false);
 
+  // Filter the space time chart curves if the first or last waypoints are hidden
+  const spaceTimeChartLayersData = useMemo(() => {
+    let filteredProjectPathTrainResult = projectPathTrainResult;
+    let filteredConflicts = conflicts;
+
+    if (!waypointsPanelData) return { filteredProjectPathTrainResult, filteredConflicts };
+
+    const { filteredWaypoints } = waypointsPanelData;
```

Comment:
Maybe we can check here that filteredWaypoints has at least 2 elements (and remove the check line 65)
```suggestion
    const { filteredWaypoints } = waypointsPanelData;
    if (filteredWaypoints.length < 2) throw new Error('filteredWaypoints has less than 2 points');
```

Reply:
We have a small gap when `filteredWaypoints` isn't updated. If we don't that, when we enter a scenario, we will have 
![Capture d’écran 2024-10-31 à 14 23 36](https://github.com/user-attachments/assets/613c3d88-0013-463e-bde2-d740e659f1ca)